crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -1512,12 +1532,26 @@ mod test {
     fn parse_use() {
         parse_all(
             use_statement(),
-            vec!["use std::hash", "use std", "use foo::bar as hello", "use bar as bar"],
+            vec![
+                "use std::hash",
+                "use std",
+                "use foo::bar as hello",
+                "use bar as bar",
+                "use foo::{bar, hello}",
+                "use foo::{bar as bar2, hello}",
+                "use foo::{bar as bar2, hello::{foo}, nested::{foo, bar}}",
+            ],
```

Comment:
Looks like `use dep::{std::println, bar::baz}` fails to parse for me because of the `dep`.

Reply:
I'm not sure if I solved it in the best way...

Reply:
I found an issue with this parsing `use dep::dep::dep;` and went to add a suggestion to fix it but the fix grew a bit so I've opened a new PR #1548. Let me know what you think.